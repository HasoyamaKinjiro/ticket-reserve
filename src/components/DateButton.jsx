import React, { forwardRef, useEffect, useState } from 'react';
import { Button } from '@mui/material';
import { useSelector } from 'react-redux';

import DateButtonStyles from '../styled/ChooseDayStyles/DateButtonStyles';

const DateButton = forwardRef(({ title, id, onClick, sx }, ref) => {
    const [isFocused, setIsFocused] = useState(false);
    const selectedDay = useSelector((state) => state.chooseDayState.selectedDay)

    useEffect(() => {
        if (id && id === selectedDay) {
            setIsFocused(true);
        } else {
            setIsFocused(false);
        }
    }, [id, selectedDay]);

    return (
        <Button
            id={id}
            color="secondary"
            variant="contained"
            sx={DateButtonStyles(isFocused, sx)}
            onClick={onClick}
            ref={ref}
        >
            {title}
        </Button>
    );
});

export default DateButton;
