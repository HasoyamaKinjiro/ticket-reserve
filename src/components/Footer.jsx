import React from 'react';
import { AppBar, Toolbar, Typography } from '@mui/material';
import { appBarFooter, toolbarFooter } from '../styled/FooterStyles';

const Footer = () => {
    return (
        <AppBar position="static" sx={appBarFooter}>
            <Toolbar sx={toolbarFooter}>
                <Typography variant="subtitle1">Â© 2024-2024 Ticket Reserve. All rights reserved.</Typography>
            </Toolbar>
        </AppBar>
    );
};

export default Footer;
