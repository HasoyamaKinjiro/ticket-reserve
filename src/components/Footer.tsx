import React from 'react';
import { AppBar, Toolbar, Typography } from '@mui/material';

import { appBarFooter, toolbarFooter } from '../styled/FooterStyles';

const Footer = () => (
    <AppBar position="static" sx={appBarFooter} role="contentinfo">
        <Toolbar sx={toolbarFooter}>
            <Typography variant="subtitle1">Â© 2024-2024 Ticket Reserve. All rights reserved.</Typography>
        </Toolbar>
    </AppBar>
);

export default Footer;
