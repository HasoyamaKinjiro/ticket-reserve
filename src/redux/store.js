import { createStore, combineReducers, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { all } from 'redux-saga/effects';

import moviesReducer, { moviesWatcher } from './ducks/movies';
import movieReducer from './ducks/reservationMovie';
import chooseDayReducer from './ducks/chooseDay';

const rootReducer = combineReducers({
    reservationMovieState: movieReducer,
    moviesState: moviesReducer,
    chooseDayState: chooseDayReducer
});

function* rootWatcher () {
    yield all([
        moviesWatcher()
    ])
}

const sagaMiddleware = createSagaMiddleware()
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(rootWatcher)

export default store;
