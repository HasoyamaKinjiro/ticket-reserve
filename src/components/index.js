import Header from './Header';
import Footer from './Footer';
import GeneralLayout from './GeneralLayout';
import DateButton from './DateButton';
import OtherDays from './OtherDays';
import ChooseDay from '../containers/ChooseDay';
import PopUpPlaceButton from './PopUpPlaceButton';
import SessionCard from './SessionCard';
import SessionPopUp from './SessionPopUp';
import Sessions from '../containers/Sessions';

export {
    Header,
    Footer,
    GeneralLayout,
    DateButton,
    OtherDays,
    PopUpPlaceButton,
    SessionCard,
    SessionPopUp,
    ChooseDay,
    Sessions
};
