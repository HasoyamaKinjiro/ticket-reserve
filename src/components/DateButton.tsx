import React, { forwardRef, MouseEventHandler, useEffect, useState } from 'react';
import { Button, SxProps, Theme } from '@mui/material';
import { useSelector } from 'react-redux';

import DateButtonStyles from '../styled/ChooseDayStyles/DateButtonStyles';

import { State } from '../redux/store';

interface DateButtonProps {
    title: string;
    id: string | undefined;
    onClick?: MouseEventHandler<HTMLButtonElement>;
    sx?: SxProps<Theme>;
}

const DateButton = forwardRef<HTMLButtonElement, DateButtonProps>(
    ({ title, id, onClick, sx }, ref) => {
    const [isFocused, setIsFocused] = useState<boolean>(false);
    const selectedDay = useSelector((state: State) => state.chooseDayState.selectedDay)

    useEffect(() => {
        if (id && id === selectedDay) {
            setIsFocused(true);
        } else {
            setIsFocused(false);
        }
    }, [id, selectedDay]);

    return (
        <Button
            id={id}
            color="secondary"
            variant="contained"
            sx={DateButtonStyles(isFocused, sx)}
            onClick={onClick}
            ref={ref}
        >
            {title}
        </Button>
    );
});

export default DateButton;
