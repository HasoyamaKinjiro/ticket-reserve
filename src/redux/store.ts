import { createStore, combineReducers, applyMiddleware } from 'redux';
import createSagaMiddleware, {  } from 'redux-saga';
import { all } from 'redux-saga/effects';

import moviesReducer, { moviesWatcher } from './ducks/movies';
import reservationMovieReducer from './ducks/reservationMovie';
import chooseDayReducer from './ducks/chooseDay';

const rootReducer = combineReducers({
    reservationMovieState: reservationMovieReducer,
    moviesState: moviesReducer,
    chooseDayState: chooseDayReducer
});

function* rootWatcher() {
    yield all([
        moviesWatcher()
    ])
}

export type State = ReturnType<typeof rootReducer>;
export type Dispatch = typeof store.dispatch;

const sagaMiddleware = createSagaMiddleware()
const store= createStore(rootReducer, applyMiddleware(sagaMiddleware) as any);
sagaMiddleware.run(rootWatcher)

export default store;




/*interface RootState {
    /!*reservationMovieState: (state: ReservationMovieState | undefined, action: ReservationMovieAction) => ReservationMovieState;
    moviesState: (state: MoviesState | undefined, action: MoviesAction) => MoviesState;*!/
    chooseDayState: (state: ChooseDayState | undefined, action: ChooseDayAction) => ChooseDayState;
}*/
