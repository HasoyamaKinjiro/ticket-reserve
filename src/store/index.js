import { createStore, combineReducers, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { rootWatcher } from '../saga';
import moviesReducer from './movies';
import movieReducer from './reservationMovie';
import chooseDayReducer from './chooseDay';

const sagaMiddleware = createSagaMiddleware()

const rootReducer = combineReducers({
    reservationMovieState: movieReducer,
    moviesState: moviesReducer,
    chooseDayState: chooseDayReducer
});

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

export default store;

sagaMiddleware.run(rootWatcher)
