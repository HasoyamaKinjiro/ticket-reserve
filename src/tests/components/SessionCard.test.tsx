import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import { SessionCard } from '../../components';

import store from '../../redux/store';

const movie = {
    "title": "Inception",
    "genre": "Sci-Fi, Thriller",
    "duration": 148,
    "releaseDate": "2010-07-16",
    "rating": 8.8,
    "description": "A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a CEO.",
    "posterUrl": "https://image.tmdb.org/t/p/w500/9gk7adHYeDvHkCSEqAvQNLV5Uge.jpg"
}

describe('SessionCard', () => {
    it('render component', () => {
        render(
            <Provider store={store}>
                <Router>
                    <SessionCard movie={movie} />
                </Router>
            </Provider>
        );

        expect(screen.getByText(/Inception/)).toBeInTheDocument();
    });

    it('open popup on click', async () => {
        render(
            <Provider store={store}>
                <Router>
                    <SessionCard movie={movie} />
                </Router>
            </Provider>
        );

        const timeBtn = screen.getByText('10:00');

        expect(timeBtn).toBeInTheDocument();

        fireEvent.click(timeBtn);
        await waitFor(() => {
            expect(store.getState().reservationMovieState.popUpIsOpen).toBe(true);
        });
    });

    it('SessionCard snapshot', () => {
        const sessionCard = render(
            <Provider store={store}>
                <Router>
                    <SessionCard movie={movie} />
                </Router>
            </Provider>
        );

        expect(sessionCard).toMatchSnapshot();
    });
});
