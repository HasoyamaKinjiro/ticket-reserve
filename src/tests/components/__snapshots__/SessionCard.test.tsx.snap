// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionCard SessionCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10o59up-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardMedia-root css-1hai4ge-MuiCardMedia-root"
            role="img"
            style="background-image: url(https://image.tmdb.org/t/p/w500/9gk7adHYeDvHkCSEqAvQNLV5Uge.jpg);"
            title="Inception"
          />
          <div
            class="MuiBox-root css-1eh63rb"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-baxsg6-MuiTypography-root"
              >
                Inception
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-fu4q7z-MuiCardActions-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-exfl4s-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    10:00
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    12:00
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    14:00
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    16:00
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    18:00
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    20:00
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10o59up-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardMedia-root css-1hai4ge-MuiCardMedia-root"
          role="img"
          style="background-image: url(https://image.tmdb.org/t/p/w500/9gk7adHYeDvHkCSEqAvQNLV5Uge.jpg);"
          title="Inception"
        />
        <div
          class="MuiBox-root css-1eh63rb"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-baxsg6-MuiTypography-root"
            >
              Inception
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-fu4q7z-MuiCardActions-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-exfl4s-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  10:00
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  12:00
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  14:00
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  16:00
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  18:00
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-5xgeot-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  20:00
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
